### PR TITLE
Jon/header

### PR DESCRIPTION
**Pull request by jonborchardt**
Had to add more code than expected to get header in

Added header
Added docs and tests for header
Added docs and test for footer (footer added previously)

footer docs: 
image

header docs:
image

### CODE REVIEW

File: components/__tests__/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -64,5 +64,11 @@ Array [
   "Typography",
   "Upload",
   "version",
+  "Columns",
```
Comment (jonborchardt) [inline components/__tests__/__snapshots__/index.test.js.snap]: .snap files are 'snapshots'
they are not code, they are a snapshot of how the dom changed due to the pr
the proper way to review, is to see if the changes are expected from the code alterations....
if they are reasonable, then we are all good

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap [hunk 2]
```diff
@@ -7127,7 +7127,7 @@ exports[`ConfigProvider components InputNumber prefixCls 1`] = `
 
 exports[`ConfigProvider components Layout configProvider 1`] = `
 <section
-  class="config-layout"
+  class="config-layout sc-bdVaJa Sxpmm"
```
Comment (jonborchardt) [inline components/config-provider/__tests__/__snapshots__/components.test.js.snap]: i dont love how styled components are messing up snapshotting... but i dont have a way around yet

File: components/columns/Columns.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import styled from 'styled-components';
+
+export interface Props {
+    count?: number;
+    gridTemplateColumns?: string;
+    breakpoint?: string;
+}
+
+export const Columns = styled.div<Props>`
```
Comment (meslater1030) [inline components/columns/Columns.tsx]: What do these columns do that the Ant `Col` does not do? Can we get some documentation around that or do you have plans to add some?

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap (hunk 2 above)
Comment (meslater1030) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap]: Maybe this guy?  https://github.com/styled-components/jest-styled-components

File: components/header/__tests__/index.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { render } from 'enzyme';
+import mountTest from '../../../tests/shared/mountTest';
+import rtlTest from '../../../tests/shared/rtlTest';
+
+import { Header, HeaderColumns, HeaderTitle } from '..';
+import { DefaultAppLayoutProvider } from '../../layout';
+import { ThemeProvider } from '../../style/themes/varnish';
+
+describe('Footer', () => {
```
Comment (meslater1030) [inline components/header/__tests__/index.test.js]: 'Header'

File: components/header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        }
+        return 0;
+    }
+
+    render() {
+        return (
+            <AppLayoutContext.Consumer>
+                {({ setHeaderHeight }) => {
+                    const offset = this.getTopOffset();
+                    setHeaderHeight(this.state.currentHeaderHeight + offset);
```
Comment (meslater1030) [inline components/header/Header.tsx]: I like this!

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap (hunk 2 above)
Comment (jonborchardt) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap]: great find!!
https://github.com/allenai/ant-design/issues/37

File: components/columns/Columns.tsx (hunk 3 above)
Comment (jonborchardt) [reply components/columns/Columns.tsx]: https://github.com/allenai/ant-design/issues/33

File: components/header/__tests__/index.test.js (hunk 4 above)
Comment (jonborchardt) [reply components/header/__tests__/index.test.js]: fixed

**Pull request merged**